server/models/contests/addModerator.js

```diff
@@ -13,30 +13,17 @@ function addModerator({ params, body, username }) {
     const { contest_id: contestId } = params
     const { moderator: moderatorUsername } = body
     pool.query(
-      `SELECT * FROM contests_moderators WHERE contest_id=? AND moderator=?`,
-      [contestId, username],
-      (error, results) => {
-        if (error) {
+      `INSERT INTO contests_moderators (contest_id, moderator) SELECT contest_id, ? FROM contests_moderators WHERE contest_id=? AND moderator=?`,
+      [moderatorUsername, contestId, username],
+      (error, res) => {
+        if (error || res === undefined) {
+          const { code } = error
+          if (code === 'ER_DUP_ENTRY') {
+            return reject('The user is already a moderator')
+          }
           return reject(error)
-        } else if (!results || !results.length) {
-          return reject(
-            'Invalid contest ID or you do not have moderator access to the contest'
-          )
         }
```

Comment:
```suggestion
        } else {
            const { affectedRows } = res
            if (!affectedRows) {
              return reject('not a moderator')
            }
        }
```

Reply:
This is needed ..It is better to do this in all endpoints where `insert select` is used

Reply:
```suggestion
        } else {
            const { affectedRows } = res
            if (!affectedRows) {
              return reject('Invalid contest ID or you do not have moderator access to the contest')
            }
        }
```
This seems better

Reply:
Don't commit this suggestion directly, I would say, it may cause some linting error..

Reply:
Yes..It will for sure..I have randomly typed